packages/material-ui-lab/src/DataGrid/DataGrid.js

```diff
@@ -346,15 +359,84 @@ const DataGrid = React.forwardRef(function DataGrid(props, ref) {
     }
   };
 
+  const [isLoading, setLoading] = React.useState(loading);
+
+  React.useEffect(() => {
```

Comment:
No need for an effect.

Reply:
If the loading prop changes  while the data grid is mounted, it will not be shown as loaded without the effect. Is that intentionally?

Reply:
Hum, this one is tricky. On one hand, we want to let users control the state, but on the other, when applying sorting or pagination, we likely want to display a loading state. What about we control it too?

Reply:
I guess we have several options:
1. 100% controlled: The loading prop is the only source of truth. This would be the easiest but will not show a loading spinner, if the built-in dataprovider is called( maybe it does not need to, since it should be fast).
2. 0% controlled: No loading prop and the waiting for the getList call will be the loading state. This would not work for empty data, since there is not way to know if it is empty or still loading.
3. 50% controlled: If the prop says ts loading, its loading ( e.g. Initial load). But if its not loading (the prop), the local loading may be applied to wait for the getList call. This will let the user show the loading state and also will show the loading state during the getList ( my favorite approve). That's the way I currently do it.
4. Modes: This would add an another prop to make the loading state completely controlled or not. This would mean, that the user would have to set the loading manually, if the provide a data provider. Also they would need to have two props in sync for the initial load of local data. The mode and the loading itself state.
I would go with 3.

Reply:
Thanks for the detailed comparison. 3. sounds like a great tradeoff. Happy to try it out. :). We can always change it later down the road if it confuses developers.